CustomerOrganizer/CustomerOrganizer/FileIO.py

```diff
@@ -52,6 +54,18 @@ def load_file(self, window):
         self.file.close()
 
 
+    # using filedialogue we can save a file with the name of the choosing of the user while also using the save_file function
+    def save_as(self, file_data, window):
+        save_window = window
+        save_window.filename = filedialog.asksaveasfilename(initialdir = "C:\\Users\\Public\\Documents", 
+                                                            title = "Select File", 
+                                                            filetypes = (("Text Files", "*.txt"), ("All Files", "*.*")))
+        self.file_location = save_window.filename
```

Comment:
Add `if not save_window.filename: return` here. After adding the `defaultextension` option, canceling will create a txt file with no name if we don't have this.